onset_maproom/calc.py

```diff
@@ -13,6 +13,49 @@ def read_zarr_data(zarr_path):
 # Growing season functions
 
 
+def estimate_sm(
+    daily_rain,
+    et,
+    taw,
+    sminit,
+    time_coord="T",
+):
+    """Estimates soil moisture from
+    rainfall,
+    evapotranspiration,
+    total available water and
+    intial soil moisture value, knowing that:
+    sm(t) = sm(t-1) + rain(t) - et(t)
+    with roof and floor respectively at taw and 0 at each time step.
+    """
+    # Get time_coord info
+    time_coord_index = daily_rain.dims.index(time_coord)
+    time_coord_size = daily_rain[time_coord].size
+    # Converts xarray array to numpy array
+    daily_rain_np = daily_rain.to_numpy()
```

Comment:
`to_numpy` was introduced in xarray 0.19.0, released in July 2021. `environment_linux.yaml` specifies `xarray=0.18.2`, so this fails for me. But `environment_macos.yaml`, having been built later, specifies `xarray=0.19.0`, so it works for you. We need to (a) find a way to keep linux and macos environments more in sync, but also (b) set up automated test runs in github so that when a developer does something that works under macos but not under linux, they're automatically notified soon after.